### ISSUE TITLE
About the batch of img size

### ISSUE THREAD

**Issue by qiaoran-dawnlight** [Visual-Behavior/detr-tensorflow#10]
Hi, thanks for the great work. The project is highly complete. And there are some different training strategy than official one. I learn some of them.

I am curious how you preprocessing the image. 

In the `get_coco_from_id`, i didnot find the resize part? 
It seems it just keep the original size?
But there is no `dataset = dataset.padded_batch(` in the dataset part.

**So I guess you already preprocess the original size to a fix size?**
cause in the config img size seems fixed and smaller `self.image_size = 376, 672`

```
def get_coco_from_id(coco_id, coco, augmentation, config, img_dir):
    # Load imag
    img = coco.loadImgs([coco_id])[0]
    # Load image
    #data_type = "train2017" if train_val == "train" else "val2017"
    filne_name = img['file_name']
    image_path = os.path.join(img_dir, filne_name) #f"{config.}/{data_type}/{filne_name}"
    image = imageio.imread(image_path)
    # Graycale to RGB if needed
    if len(image.shape) == 2: image = gray2rgb(image)
    # Retrieve the image label
    t_bbox, t_class, is_crowd = get_coco_labels(coco, img['id'], image.shape, augmentation)
    # Apply augmentations
    if len(t_bbox) > 0 and augmentation is not None:
        image, t_bbox, t_class = transformation.detr_transform(image, t_bbox,  t_class, config, augmentation)
    # Normalized images
    image = processing.normalized_images(image, config)
    # Set type for tensorflow        
    image = image.astype(np.float32)
    t_bbox = t_bbox.astype(np.float32)
    t_class = t_class.astype(np.int64)
    return image, t_bbox, t_class, is_crowd
```

```
    # Setup the data pipeline
    img_ids = coco.getImgIds()
    shuffle(img_ids)
    dataset = tf.data.Dataset.from_tensor_slices(img_ids)
    # Shuffle the dataset
    dataset = dataset.shuffle(1000)
    # Retrieve img and labels
    outputs_types=(tf.float32, tf.float32, tf.int64, tf.int64)
    dataset = dataset.map(lambda idx: processing.numpy_fc(
        idx, get_coco_from_id, outputs_types=outputs_types, coco=coco, augmentation=augmentation, config=config, img_dir=img_dir)
    , num_parallel_calls=tf.data.experimental.AUTOTUNE)
    dataset = dataset.filter(lambda imgs, tbbox, tclass, iscrowd: tf.shape(tbbox)[0] > 0 and iscrowd != 1)
    dataset = dataset.map(lambda imgs, tbbox, tclass, iscrowd: (imgs, tbbox, tclass), num_parallel_calls=tf.data.experimental.AUTOTUNE)
    
    # Pad bbox and labels
    dataset = dataset.map(processing.pad_labels, num_parallel_calls=tf.data.experimental.AUTOTUNE)

    dataset = dataset.batch(batch_size, drop_remainder=True)
    dataset = dataset.prefetch(32)
    
```

**Comment by thibo73800** [comment]
Hi,

Thanks for your feedback. Indeed, currently, the repository uses a fixed-size image-size to train the model. 

It is for a first implementation a more memory-efficient approach than padding all images to a larger size and lead to similar results (on the target size) if we have in mind a target image size for the model deployment.

This differs from the PyTorch repository that trains on multiple image sizes. However, if this is an important feature for you, we can think about adding this feature to the pipelines. Let us know about your use cases !

Best,
Thibault

**Comment by qiaoran-dawnlight** [comment]
Thanks fo the quick explanation. And appreciate it for adding the feature.
But i am doing my own DETR implantation in a non-img dataset. 
So i mostly borrow the model part.  
Personally, i more look forward some latest model implementation, like DEFORMABLE DETR. 
Thanks again for you sharing. 
Looking for your continue updating.
